From 8652be0c270d480ecb3d56deeeace4fb936cdd65 Mon Sep 17 00:00:00 2001
From: Yuvarani V <quic_yuvarani@quicinc.com>
Date: Fri, 24 Nov 2023 18:08:10 +0530
Subject: [PATCH] ath12k: Set critical update params on beacon template

Change to update category 1 and category 2 critical update params
on WMI_TAG_BCN_TMPL_CMD

Signed-off-by: Rathees Kumar R Chinannan <quic_rrchinan@quicinc.com>
Signed-off-by: Yuvarani V <quic_yuvarani@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.h | 23 -----------------------
 drivers/net/wireless/ath/ath12k/wmi.c |  8 ++++----
 2 files changed, 4 insertions(+), 27 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.h b/drivers/net/wireless/ath/ath12k/mac.h
index dec6eb4..3d22fe3 100644
--- a/drivers/net/wireless/ath/ath12k/mac.h
+++ b/drivers/net/wireless/ath/ath12k/mac.h
@@ -88,29 +88,6 @@ enum ath12k_supported_bw {
 #define ATH12K_PDEV_TX_POWER_INVALID	(-1)
 #define ATH12K_PDEV_TX_POWER_REFRESH_TIME_MSECS 5000 /* msecs */
 
-/* Critical Update Category 1 : Inclusion of new IEs */
-#define ATH12K_CRITICAL_UPDATE_CAT_1  (IEEE80211_CU_INCLUDE_CSA_ELEM | \
-				       IEEE80211_CU_INCLUDE_ECSA_ELEM | \
-				       IEEE80211_CU_INCLUDE_QUIET_ELEM | \
-				       IEEE80211_CU_INCLUDE_WBCS_ELEM | \
-				       IEEE80211_CU_INCLUDE_CSW_ELEM | \
-				       IEEE80211_CU_INCLUDE_OMN_ELEM | \
-				       IEEE80211_CU_INCLUDE_Q_CHAN_ELEM | \
-				       IEEE80211_CU_INCLUDE_B_TWT_ELEM |\
-				       IEEE80211_CU_INCLUDE_B_TWT_EXIST_ELEM | \
-				       IEEE80211_CU_INCLUDE_BCCA_ELEM)
-
-/* Critical Update Category 2 : Modification of existing IEs */
-#define ATH12K_CRITICAL_UPDATE_CAT_2  (IEEE80211_CU_MODIFY_EDCA_PARAM_ELEM | \
-				       IEEE80211_CU_MODIFY_DSSS_PARAM_ELEM | \
-				       IEEE80211_CU_MODIFY_HT_OPER_ELEM | \
-				       IEEE80211_CU_MODIFY_VHT_OPER_ELEM | \
-				       IEEE80211_CU_MODIFY_HE_OPER_ELEM | \
-				       IEEE80211_CU_MODIFY_MU_EDCA_PARAM_ELEM | \
-				       IEEE80211_CU_MODIFY_SR_PARAM_ELEM | \
-				       IEEE80211_CU_MODIFY_UORA_PARAM_ELEM | \
-				       IEEE80211_CU_MODIFY_EHT_OPER_ELEM)
-
 enum ath12k_background_dfs_events {
 	ATH12K_BGDFS_SUCCESS,
 	ATH12K_BGDFS_ABORT,
diff --git a/drivers/net/wireless/ath/ath12k/wmi.c b/drivers/net/wireless/ath/ath12k/wmi.c
index 84923d0..70aef9e 100644
--- a/drivers/net/wireless/ath/ath12k/wmi.c
+++ b/drivers/net/wireless/ath/ath12k/wmi.c
@@ -2748,7 +2748,6 @@ static void ath12k_wmi_bcn_fill_ml_info(struct ath12k_link_vif *arvif,
 {
 	struct ath12k_base *ab = arvif->ar->ab;
 	struct ieee80211_bss_conf *link_conf;
-	u32 cu_flags;
 	u32 vdev_id = arvif->vdev_id;
 	u64 vdev_map_cat1 = 0;
 	u64 vdev_map_cat2 = 0;
@@ -2761,7 +2760,6 @@ static void ath12k_wmi_bcn_fill_ml_info(struct ath12k_link_vif *arvif,
 		goto err_fill_ml_info;
 	}
 
-	cu_flags = link_conf->critical_update_flag;
 	rcu_read_unlock();
 
 	ml_info->tlv_header = ath12k_wmi_tlv_cmd_hdr(WMI_TAG_BCN_TMPL_ML_INFO_CMD,
@@ -2771,13 +2769,15 @@ static void ath12k_wmi_bcn_fill_ml_info(struct ath12k_link_vif *arvif,
 	/* if this is cu cat 1 for tx vdev, then it applies to all
 	 * non-tx vdevs as well set all non-tx mld's vdev id in bitmap
 	 */
-	if (cu_flags & ATH12K_CRITICAL_UPDATE_CAT_1) {
+	if (link_conf->elemid_added) {
 		set_bit(vdev_id, &vdev_map_cat1);
 		vdev_map_cat1 |= non_tx_ml_vdev_bmap;
 	}
 
-	if (cu_flags & ATH12K_CRITICAL_UPDATE_CAT_2)
+	if (link_conf->elemid_modified) {
 		set_bit(vdev_id, &vdev_map_cat2);
+		vdev_map_cat2 |= non_tx_ml_vdev_bmap;
+	}
 
 err_fill_ml_info:
 	ml_info->cu_vdev_map_cat1_lo =
-- 
2.17.1

